### PR TITLE
[WIP] ZNCLDJ11LM direction support

### PR DESCRIPTION
**Pull request by yozik04**
Adding for easy viewing. Will rewrite to make it into one commit and without debug messaging in the end.

Original issue https://github.com/Koenkk/zigbee2mqtt/issues/1639

### CODE REVIEW

**Comment by Koenkk** [comment]
Hi, whats the current status?

**Comment by yozik04** [comment]
Waiting for your reply. Sent you a dump to email. Z2m sends different packet from required one.

**Comment by Koenkk** [comment]
I see that indeed somehow the manufacturer code is missing, did some debugging but cannot find why. Need to check after my holidays (don't have more time now unfortunately).

**Comment by yozik04** [comment]
No worries. Take your time. Thank you for looking into it.

**Comment by Koenkk** [comment]
The issue should be fixed now, the issue was the `options` was redefined, leading to `options.xiaomi` being null.

I've added further review comments, if that has been checked and this works please make a merge request for the documentation (https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/devices/ZNCLDJ11LM.md) and this can be merged.

File: converters/fromZigbee.js [hunk 1]
```diff
@@ -1201,11 +1201,22 @@ const converters = {
             }
         },
     },
+    ZNCLDJ11LM_ZNCLDJ12LM_curtain_options_output: {
+        cluster: 'genBasic',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options) => {
+            console.log(`genBasic ${JSON.stringify(msg.data)}`);
+            if (msg.data['1025']) {
+                return {options: msg.data['1025']}
```
Comment (Koenkk) [inline converters/fromZigbee.js]: Please convert this to something human friendly like https://github.com/Koenkk/zigbee-herdsman-converters/pull/835/files#diff-3e0c275bd41af699153bf59d8dda4a2aR762

File: converters/fromZigbee.js [hunk 2]
```diff
@@ -1201,11 +1201,22 @@ const converters = {
...
+            }
+        },
+    },
     ZNCLDJ11LM_ZNCLDJ12LM_curtain_analog_output: {
         cluster: 'genAnalogOutput',
         type: ['attributeReport', 'readResponse'],
         convert: (model, msg, publish, options) => {
             let running = false;
+            console.log(`genAnalogOutput ${JSON.stringify(msg.data)}`);
```
Comment (Koenkk) [inline converters/fromZigbee.js]: Remove this

File: converters/toZigbee.js [hunk 3]
```diff
@@ -746,6 +746,33 @@ const converters = {
...
+                0x07,
+                0x00,
+                opts.reset_move ? 0x01: 0x02,
+                0x00,
+                opts.direction == 'reverse' ? 0x01: 0x00,
+                0x04,
+                !opts.hand_open ? 0x01: 0x00,
+                0x12,
+            ];
+            meta.logger.info(`ZNCLDJ11LM convertSet ${JSON.stringify(payload)}`);
```
Comment (Koenkk) [inline converters/toZigbee.js]: This should log something human friendly, otherwise change level to `meta.logger.debug`.

**Comment by yozik04** [comment]
> The issue should be fixed now, the issue was the `options` was redefined, leading to `options.xiaomi` being null.
OMG, indeed... Sorry...

**Comment by yozik04** [comment]
This part is ready and tested with my ZNCLDJ11LM on latest dev. Will add some documentation a bit later.

**Comment by yozik04** [comment]
Done.

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**